### PR TITLE
fix: provenance build type v2

### PR DESCRIPTION
**Pull request by bdehamer**
Various updates to the generated provenance statement:

* Removes forgeable provenance fields
* Updates the `materials.uri` value to include the `GITHUB_REF` value
* Updates the `entryPoint` value to include just the repo-relative path to the initiating workflow:\
`github/foo/.github/workflows/publish.yml@refs/heads/main` → `.github/workflows/publish.yml`
* Updates the `builder.id` and `buildType` fields

### CODE REVIEW

File: workspaces/libnpmpublish/lib/provenance.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ const BUILD_TYPE_VERSION = 'v1'
 
 const generateProvenance = async (subject, opts) => {
   const { env } = process
+  const [workflowPath] = (env.GITHUB_WORKFLOW_REF || '')
+    .replace(env.GITHUB_REPOSITORY + '/', '')
+    .split('@')
```
Comment (ljharb) [inline workspaces/libnpmpublish/lib/provenance.js:16]: are there tests that explain why this change is needed?

**Comment by bdehamer** [comment]
@feelepxyz thoughts on this last commit which updates the `builder.id` and `buildType`

* `builder.id` changes from `https://github.com/npm/cli@9.4.0` to `https://github.com/actions/runner` (to be augmented with actual runner type and runner version at some later date)
* `buildType` remains the same but we increment the version from `v1` to `v2` and change the "@" to "/" to ensure URI parseability: `https://github.com/npm/cli/gha/v2`

File: workspaces/libnpmpublish/lib/provenance.js [hunk 2]
```diff
@@ -4,39 +4,40 @@ const INTOTO_PAYLOAD_TYPE = 'application/vnd.in-toto+json'
...
 
-const BUILDER_ID_PREFIX = 'https://github.com/npm/cli'
+const BUILDER_ID = 'https://github.com/actions/runner'
 const BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
-const BUILD_TYPE_VERSION = 'v1'
+const BUILD_TYPE_VERSION = 'v2'
 
 const generateProvenance = async (subject, opts) => {
   const { env } = process
+  /* istanbul ignore next */
```
Comment (wraithgar) [inline workspaces/libnpmpublish/lib/provenance.js]: Needs a test or a comment explaining why it's not tested

**Comment by wraithgar** [comment]
What does a typical `GITHUB_WORKFLOW_REF` look like?

File: workspaces/libnpmpublish/lib/provenance.js (hunk 2 above)
Comment (bdehamer) [reply workspaces/libnpmpublish/lib/provenance.js]: Just a placeholder until I had the test in place. Removed.

**Comment by bdehamer** [comment]
> What does a typical `GITHUB_WORKFLOW_REF` look like?

The raw `GITHUB_WORKFLOW_REF` looks something like:

```
github/package-security/.github/workflows/dump-env.yml@refs/heads/main
```

and we need to extract just the repo-relative workflow path from that:

```
.github/workflows/dump-env.yml
```

**Comment by bdehamer** [comment]
With these changes, the new provenance statement will look something like:

```json
{
  "_type": "https://in-toto.io/Statement/v0.1",
  "subject": [
    {
      "name": "pkg:npm/sigstore@1.0.0",
      "digest": {
        "sha512": "7bea9f6e7ff37f5fab0b36bf061200fff03099fd2fd696b91d04bc5e4f225eb9fd6e0cadcad54ba980f43fb352a99e8810b4e0abeb5c0ef2cf9108cd1f258b36"
      }
    }
  ],
  "predicateType": "https://slsa.dev/provenance/v0.2",
  "predicate": {
    "buildType": "https://github.com/npm/cli/gha/v2",
    "builder": {
      "id": "https://github.com/actions/runner"
    },
    "invocation": {
      "configSource": {
        "uri": "git+https://github.com/sigstore/sigstore-js@refs/tags/v1.0.0",
        "digest": {
          "sha1": "06528997c3c8ab9f864fad9a3658446aca7fd86d"
        },
        "entryPoint": ".github/workflows/publish.yml"
      },
      "parameters": {},
      "environment": {
        "GITHUB_EVENT_NAME": "release",
        "GITHUB_REF": "refs/tags/v1.0.0",
        "GITHUB_REPOSITORY": "sigstore/sigstore-js",
        "GITHUB_REPOSITORY_ID": "495574555",
        "GITHUB_REPOSITORY_OWNER_ID": "71096353",
        "GITHUB_RUN_ATTEMPT": "1",
        "GITHUB_RUN_ID": "4137028816",
        "GITHUB_SHA": "06528997c3c8ab9f864fad9a3658446aca7fd86d",
        "GITHUB_WORKFLOW_REF": "sigstore/sigstore-js/.github/workflows/publish.yml@refs/tags/v1.0.0",
        "GITHUB_WORKFLOW_SHA": "06528997c3c8ab9f864fad9a3658446aca7fd86d"
      }
    },
    "metadata": {
      "buildInvocationId": "4137028816-1",
      "completeness": {
        "parameters": false,
        "environment": false,
        "materials": false
      },
      "reproducible": false
    },
    "materials": [
      {
        "uri": "git+https://github.com/sigstore/sigstore-js@refs/tags/v1.0.0",
        "digest": {
          "sha1": "06528997c3c8ab9f864fad9a3658446aca7fd86d"
        }
      }
    ]
  }
}
```

File: workspaces/libnpmpublish/test/publish.js [hunk 3]
```diff
@@ -732,7 +792,12 @@ t.test('publish existing package with provenance in gha', async t => {
 
   registry.getVisibility({ spec, visibility: { public: true } })
   registry.nock.put(`/${spec.escapedName}`, body => {
-    return t.match(body, packument, 'posted packument matches expectations')
+    const bundleAttachment = body._attachments['@npmcli/libnpmpublish-test-1.0.0.sigstore']
+    const bundle = JSON.parse(bundleAttachment.data)
+    const provenance = JSON.parse(Buffer.from(bundle.dsseEnvelope.payload, 'base64').toString())
+
+    return t.match(body, packument, 'posted packument matches expectations') &&
```
Comment (wraithgar) [inline workspaces/libnpmpublish/test/publish.js]: I think two `t.match` statements followed by a `return true` match more closely how we do it elsewhere.

Comment (wraithgar) [reply workspaces/libnpmpublish/test/publish.js]: Actually @nlf reminds me that `t.match` has very unexpected behavior. `t.hasStrict` is more appropriate.

Comment (bdehamer) [reply workspaces/libnpmpublish/test/publish.js]: Switched to `t.hasStrict` and pared things down a bit so I'm only looking at the important bits of the provenance.

**Review by wraithgar — APPROVED** [review]

File: workspaces/libnpmpublish/test/publish.js [hunk 4]
```diff
@@ -732,7 +762,24 @@ t.test('publish existing package with provenance in gha', async t => {
...
+    t.hasStrict(body, packument, 'posted packument matches expectations')
+    t.hasStrict(provenance.subject[0],
+      expectedSubject,
+      'provenance subject matches expectations')
+    t.hasStrict(provenance.predicate.buildType,
+      'https://github.com/npm/cli/gha/v2',
+      'provenance subject matches expectations')
+    t.hasStrict(provenance.predicate.builder.id,
+      'https://github.com/actions/runner',
+      'provenance subject matches expectations')
```
Comment (feelepxyz) [inline workspaces/libnpmpublish/test/publish.js]:
```suggestion
      'buildType matches expectations')
    t.hasStrict(provenance.predicate.builder.id,
      'https://github.com/actions/runner',
      'builder id matches expectations')
```

**Review by feelepxyz — APPROVED** [review]
Minor spec copy, otherwise LGTM 👍
